### DISCUSSION TITLE
Search in TN stateparts.

### DISCUSSION THREAD

**Discussion by Carolina-Camping-Couple** [juftin/camply#40] [Q&A] [answered]
Hi, camply is such a great tool to book a campground. I I was wondering if it could be used to search state parks as well?  TN has some wonderful parks.

https://gis.tnstateparks.com/datasets/tn-state-parks-points/api

**Comment by juftin** [comment, marked as answer]
Hey @Carolina-Camping-Couple thanks! `camply` is absolutely capable of expanding to other campsite booking services. I built it with the concept of being able to support different ["providers"](https://github.com/juftin/camply/tree/main/camply/providers) - currently it supports RecreationDotGov and Xanterra.

Here comes the unfortunate part 😢. Standing up a new provider (i.e. "TN State Parks") requires quite a bit of exploration, experimentation, and heads down coding time to actually implement with `camply`. Each one of these provider modules needs to define how to interact with their respective API and perform actions like `List Campgrounds / Campsites`, `Find Availabilities`.  I'm more than happy to help with the implementation but the discovery is the tough part.

 If you do want to contribute to `camply` by adding a new provider let me know and I'm happy to give you some more details of what the implementation would require. Thanks!
